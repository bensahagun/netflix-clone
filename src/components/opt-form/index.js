import React from 'react';
import { Container, Input, Button, Break, Text } from './styles/opt-form';

export default function OptForm({ children, ...restProps }) {
  return <Container {...restProps}>{children}</Container>;
}

OptForm.Input = function ({ ...restProps }) {
  return <Input {...restProps}></Input>;
};

OptForm.Button = function ({ children, ...restProps }) {
  return (
    <Button {...restProps}>
      {children} <img src='/images/icons/chevron-right.png' alt='Try now' />
    </Button>
  );
};

OptForm.Text = function ({ children, ...restProps }) {
  return <Text {...restProps}>{children}</Text>;
};

OptForm.Break = function ({ ...restProps }) {
  return <Break {...restProps} />;
};
