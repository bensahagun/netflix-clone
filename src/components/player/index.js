import React, { useState, useContext, createContext } from 'react';
import ReactDOM from 'react-dom';
import { Container, Button, FeatureButton, Overlay, Inner, Close } from './styles/player';

export const PlayerContext = createContext();

export default function Player({ children, ...restProps }) {
  const [showPlayer, setShowPlayer] = useState(false);

  return (
    <PlayerContext.Provider value={{ showPlayer, setShowPlayer }}>
      <Container {...restProps}>{children}</Container>
    </PlayerContext.Provider>
  );
}

Player.Video = function PlayerVideo({ src, ...restProps }) {
  const { showPlayer, setShowPlayer } = useContext(PlayerContext);

  return showPlayer
    ? ReactDOM.createPortal(
        <Overlay onClick={() => setShowPlayer(false)} {...restProps}>
          <Inner>
            <video id='netflix-player' controls autoPlay='autoplay'>
              <source src={src} type='video/mp4' />
            </video>
          </Inner>
          <Close />
        </Overlay>,
        document.body
      )
    : null;
};

Player.Button = function PlayerButton({ ...restProps }) {
  // eslint-disable-next-line no-unused-vars
  const { showPlayer, setShowPlayer } = useContext(PlayerContext);

  return (
    <Button {...restProps} onClick={() => setShowPlayer((showPlayer) => !showPlayer)}>
      Play
    </Button>
  );
};

Player.FeatureButton = function PlayerFeatureButton({ ...restProps }) {
  // eslint-disable-next-line no-unused-vars
  const { showPlayer, setShowPlayer } = useContext(PlayerContext);

  return (
    <FeatureButton {...restProps} onClick={() => setShowPlayer((showPlayer) => !showPlayer)}>
      Play Now
    </FeatureButton>
  );
};
