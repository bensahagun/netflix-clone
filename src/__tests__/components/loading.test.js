import React from 'react';
import { getByTestId, render } from '@testing-library/react';
import { Loading } from '../../components';

describe('<Loading/>', () => {
  it('render the <Loading/> component', () => {
    const { container, getByTestId } = render(<Loading src='/images/ben.png' data-testid='loading' />);

    expect(getByTestId('loading')).toBeTruthy();
    expect(getByTestId('loading-picture')).toBeTruthy();
    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders the <Loading.ReleaseBody/>', () => {
    const { container } = render(<Loading.ReleaseBody />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
