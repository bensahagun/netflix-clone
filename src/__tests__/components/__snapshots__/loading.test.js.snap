// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loading/> render the <Loading/> component 1`] = `
<div
  class="sc-bdfBwQ ebhJOW"
  data-testid="loading"
>
  <img
    class="sc-gsTCUz bobqpP"
    data-testid="loading-picture"
    src="/images/users//images/ben.png.png"
  />
</div>
`;

exports[`<Loading/> renders the <Loading.ReleaseBody/> 1`] = `null`;
